pkgs/development/libraries/ogre/1.11.x.nix

```
@@ -0,0 +1,45 @@
+{ fetchFromGitHub, stdenv, lib
+, cmake, libGLU, libGL
+, freetype, freeimage, zziplib, xorgproto, libXrandr
+, libXaw, freeglut, libXt, libpng, boost, ois
+, libX11, libXmu, libSM, pkg-config
+, libXxf86vm, libICE
+, libXrender
+, withNvidiaCg ? false, nvidia_cg_toolkit
+, withSamples ? false }:
+
+stdenv.mkDerivation rec {
+  pname = "ogre";
+  version = "1.11.6";
+
+  src = fetchFromGitHub {
+    owner = "OGRECave";
+    repo = "ogre";
+    rev = "v${version}";
+    sha256 = "lCRa3sloF0o7Wz8QAuPyJTXYrvNmAfuKsz/DzjgSyls=";
+  };
+
+  cmakeFlags = [ "-DOGRE_BUILD_SAMPLES=${toString withSamples}" ]
+    ++ map (x: "-DOGRE_BUILD_PLUGIN_${x}=on")
+           ([ "BSP" "OCTREE" "PCZ" "PFX" ] ++ lib.optional withNvidiaCg "CG")
+    ++ map (x: "-DOGRE_BUILD_RENDERSYSTEM_${x}=on") [ "GL" ];
+
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs =
+   [ libGLU libGL
+     freetype freeimage zziplib xorgproto libXrandr
+     libXaw freeglut libXt libpng boost ois
+     libX11 libXmu libSM
+     libXxf86vm libICE
+     libXrender
+   ] ++ lib.optional withNvidiaCg nvidia_cg_toolkit;
+
+  meta = {
+    description = "A 3D engine";
+    homepage = "https://www.ogre3d.org/";
+    maintainers = [ lib.maintainers.raskin ];
+    platforms = lib.platforms.linux;
+    license = lib.licenses.mit;
```

```suggestion
  meta = with lib; {
    description = "A 3D engine";
    homepage = "https://www.ogre3d.org/";
    maintainers = [ maintainers.raskin ];
    platforms = platforms.linux;
    license = licenses.mit;
```
Also please only add yourself as a maintainer.